sanitize_test.go

```diff
@@ -319,6 +319,36 @@ func TestEmptyAttributes(t *testing.T) {
 	}
 }
 
+func TestDataAttributes(t *testing.T) {
+
+	p := UGCPolicy()
+	p.AllowDataAttributes()
+
+	tests := []test{
+		{
+			in:       `<p data-cfg="dave">text</p>`,
+			expected: `<p data-cfg="dave">text</p>`,
+		},
+		{
+			in:       `<p data-component="dave">text</p>`,
+			expected: `<p data-component="dave">text</p>`,
+		},
```

Comment:
Add a test that fails, spec says data-* attributes cannot be upper case, so that's a good one.

Reply:
I’ll do that ASAP cheers!

Reply:
@buro9 a test I haven't pushed yet for the uppercasing is failing as passing in ```data-UPCASE``` gives ```data-upcase``` in the output, is some normalising happening elsewhere? If so the uppercase data attributes will be taken care of? Added tests for the other parts of the spec, namely no semi-colons and no ```data-xml-``` prefixes.

Reply:
@buro9 yep looking at ```https://godoc.org/golang.org/x/net/html``` looks like the lowercasing is done when the ```html.Attribute``` is created.

Reply:
I suspected that might happen but hoped it wouldn't. Damn... OK, it's good without that :)